server/socket/socketHelperFunctions.js

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable no-unused-vars */
+require('express');
+const mongoose = require('mongoose');
+const ssotToRobotParser = require('../services/SsotToRobotParsing/SsotToRobotParser.js');
+const ssotModels = require('../models/singleSourceOfTruthModel.js');
+const userAccessObject = require('../models/userAccessObjectModel.js');
+const jobsModel = require('../models/robotJobModel.js');
+
+mongoose.set('useFindAndModify', false);
+
+/**
+ * @description Fetches the ssot of a given robot from the database and parses the ssot to robot code
+ * @param {String} robotId the id of the robot we want the robot code for
+ */
+exports.getRobotCode = async (robotId) => {
+  try {
+    mongoose.connect(process.env.MONGODB_URI, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true,
+    });
+    const ssot = await mongoose.model('SSoT').findById(robotId).exec();
+    const robotCode = ssotToRobotParser.parseSsotToRobotCode(ssot);
+    return robotCode;
+  } catch (err) {
+    return console.error(err);
+  }
+};
+
+/**
+ * @description Fetches the ssot from the database and parses the ssot to robot code
+ * @param {String} robotId the id of the robot we want the robot code for
+ */
+exports.getAllUserIds = async () => {
+  try {
+    mongoose.connect(process.env.MONGODB_URI, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true,
+    });
+    const userObjs = await mongoose.model('userAccessObject').find();
+    let userIds = [];
+    if (userObjs.length > 0) {
+      userIds = userObjs.map((obj) => String(obj.userId));
+    }
+    return userIds;
+  } catch (err) {
+    return console.error(err);
+  }
+};
+
+/**
+ * @description Creates a Job in the database for a robot execution of a specific user
+ * @param {String} userId the id of the user that wants to execute a robot
+ * @param {String} robotId the id of the robot to be executed
+ * @param {String} status the current status of the job (either waiting, executing, success or failed)
+ * @param {Array} parameters different parameters the user defined before executing the robot
+ */
+exports.createJob = async (userId, robotId, status, parameters) => {
+  const job = new jobsModel.Job({
```

Comment:
you could also use
```
const job = new mongoose.model('Job')({paramaterObject})
```
as constructor. Then you do not need the export statement to a variable in JobModel. But I'm not sure if thats better, it can stay that way IMO